dash/include/dash/Team.h

```diff
@@ -76,12 +77,12 @@ class Team
 
   inline iterator begin()
   {
-    return iterator(0);
+    return {0};
   }
 
   inline iterator end()
   {
-    return iterator(size());
+    return {static_cast<int>(size())};
```

Comment:
This construction looks a little bit unintuitive.

Reply:
why? It is actually the same "problem" as explained below.